### ISSUE TITLE
💅 False positive for `lint/suspicious/noEmptyBlockStatements` with constructor

### ISSUE THREAD

**Issue by igorlfs** [biomejs/biome#3005]
### Environment information

```bash
CLI:
  Version:                      1.7.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.13.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        complexity/noBannedTypes = "error"
                                complexity/noExtraBooleanCast = "error"
                                complexity/noForEach = "off"
                                complexity/noMultipleSpacesInRegularExpressionLiterals = "error"
                                complexity/noStaticOnlyClass = "off"
                                complexity/noUselessCatch = "error"
                                complexity/noUselessConstructor = "error"
                                complexity/noUselessThisAlias = "error"
                                complexity/noUselessTypeConstraint = "error"
                                complexity/noVoid = "error"
                                complexity/noWith = "error"
                                complexity/useArrowFunction = "off"
                                correctness/noConstAssign = "error"
                                correctness/noConstantCondition = "error"
                                correctness/noEmptyCharacterClassInRegex = "error"
                                correctness/noEmptyPattern = "error"
                                correctness/noGlobalObjectCalls = "error"
                                correctness/noInvalidConstructorSuper = "error"
                                correctness/noInvalidNewBuiltin = "error"
                                correctness/noNonoctalDecimalEscape = "error"
                                correctness/noPrecisionLoss = "error"
                                correctness/noSelfAssign = "error"
                                correctness/noSetterReturn = "error"
                                correctness/noSwitchDeclarations = "error"
                                correctness/noUndeclaredVariables = "error"
                                correctness/noUnreachable = "error"
                                correctness/noUnreachableSuper = "error"
                                correctness/noUnsafeFinally = "error"
                                correctness/noUnsafeOptionalChaining = "error"
                                correctness/noUnusedLabels = "error"
                                correctness/noUnusedPrivateClassMembers = "error"
                                correctness/noUnusedVariables = "error"
                                correctness/useIsNan = "error"
                                correctness/useValidForDirection = "error"
                                correctness/useYield = "error"
                                style/noCommaOperator = "error"
                                style/noNamespace = "error"
                                style/noNonNullAssertion = "off"
                                style/noParameterAssign = "error"
                                style/noUselessElse = "off"
                                style/useAsConstAssertion = "error"
                                style/useBlockStatements = "off"
                                style/useConsistentArrayType = {"level":"error","options":{"syntax":"generic"}}
                                style/useForOf = "error"
                                style/useNumericLiterals = "error"
                                style/useTemplate = "error"
                                suspicious/noAssignInExpressions = "error"
                                suspicious/noAsyncPromiseExecutor = "error"
                                suspicious/noCatchAssign = "error"
                                suspicious/noClassAssign = "error"
                                suspicious/noCompareNegZero = "error"
                                suspicious/noConfusingLabels = "error"
                                suspicious/noConsoleLog = "error"
                                suspicious/noControlCharactersInRegex = "error"
                                suspicious/noDebugger = "error"
                                suspicious/noDoubleEquals = "error"
                                suspicious/noDuplicateCase = "error"
                                suspicious/noDuplicateClassMembers = "error"
                                suspicious/noDuplicateObjectKeys = "error"
                                suspicious/noDuplicateParameters = "error"
                                suspicious/noEmptyBlockStatements = "error"
                                suspicious/noExplicitAny = "error"
                                suspicious/noExtraNonNullAssertion = "error"
                                suspicious/noFallthroughSwitchClause = "error"
                                suspicious/noFunctionAssign = "error"
                                suspicious/noGlobalAssign = "error"
                                suspicious/noImportAssign = "error"
                                suspicious/noMisleadingCharacterClass = "error"
                                suspicious/noMisleadingInstantiator = "error"
                                suspicious/noPrototypeBuiltins = "error"
                                suspicious/noRedeclare = "error"
                                suspicious/noShadowRestrictedNames = "error"
                                suspicious/noUnsafeDeclarationMerging = "error"
                                suspicious/noUnsafeNegation = "error"
                                suspicious/useGetterReturn = "error"
                                suspicious/useValidTypeof = "error"

Workspace:
  Open Documents:               0
```

### Rule name

lint/suspicious/noEmptyBlockStatements

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZQB4AHAAbwByAHQAIABjAGwAYQBzAHMAIABGAG8AbwBCAGEAcgAgAHsACgAgACAAIAAgAGMAbwBuAHMAdAByAHUAYwB0AG8AcgAoAAoAIAAgACAAIAAgACAAIAAgAHAAcgBpAHYAYQB0AGUAIABmAG8AbwA6ACAAcwB0AHIAaQBuAGcALAAKACAAIAAgACAAIAAgACAAIABwAHIAaQB2AGEAdABlACAAYgBhAHIAOgAgAHMAdAByAGkAbgBnACwACgAgACAAIAAgACkAIAB7AH0ACgAKACAAIABwAHUAYgBsAGkAYwAgAGIAYQB6ACgAKQAgAHsACgAgACAAIAAgAHQAaABpAHMALgBmAG8AbwAgAD0AIAB0AGgAaQBzAC4AYgBhAHIACgAgACAAfQAKAH0A

### Expected result

Biome shouldn't trigger the rule, since it's just a shorthand for class initialization

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
I've never seen this syntax before, but looking at the docs for this rule, this is a bug. I'd like to pick this up.
